File: temp/py/segregate0and1.py
def segregate(nums):
    left = 0
    right = len(nums) - 1
    i = 0

    while i <= right:
        if nums[i] == 0:
            swap(nums, i, left)
            left += 1
            i += 1
        else:
            swap(nums, i, right)
            right -= 1
    
def swap(nums, i, j):
    nums[i] = nums[i] ^ nums[j]
    nums[j] = nums[i] ^ nums[j]
    nums[i] = nums[i] ^ nums[j]


arr = list(map(int, input().split(",")))
segregate(arr)
print(arr)

File: Two_Pointers/Segregate0and1.java
package two_pointers;

import java.util.Scanner;

// https://www.geeksforgeeks.org/problems/segregate-0s-and-1s5106/1

public class Segregate0and1{
    static void segregate0and1(int[] arr) {
        int left = 0, right = arr.length-1;
        int i = 0;
        while(i <= right){
            if(arr[i] == 0){
                swap(arr, i, left);
                left++;
                i++;
            }else{
                swap(arr, i, right);
                right--;
            }
        }
    }
    static void swap(int nums[], int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        segregate0and1(arr);
        System.out.println(arr.toString());

        sc.close();
    }
}